src/models/modules/image_encoder.py

```diff
@@ -112,14 +112,16 @@ def forward(self, image_tensor: torch.Tensor) -> Any:
 class VGGEncoder(nn.Module):
     """Pre Trained VGG Encoder Module"""
 
-    def __init__(self) -> None:
+    def __init__(self, device: Any) -> None:
```

Comment:
`nn.Module` class is be default has a functionality for moving weights between devices. If another `nn.Module` is getting attached to base class through `self.` then when you apply `.to(device)` every weight part moves to that device. There's no reason to hardcode moving var `model`. You can just init it like `self.model = torch.hub.load.<bla-bla>` and you won't need to pass `device` argument

Reply:
changed this, thanks for telling me this. Didn't know about it.